python_modules/dagster-webserver/dagster_webserver/graphql.py

```diff
@@ -241,18 +242,24 @@ async def execute_graphql_request(
         request_context = self.make_request_context(request)
 
         def _graphql_request():
-            return run(
-                self._graphql_schema.execute_async(
-                    query,
-                    variables=variables,
-                    operation_name=operation_name,
-                    context=request_context,
-                    middleware=self._graphql_middleware,
+            with self.wrap_graphql_execution(request):
+                return run(
+                    self._graphql_schema.execute_async(
+                        query,
+                        variables=variables,
+                        operation_name=operation_name,
+                        context=request_context,
+                        middleware=self._graphql_middleware,
+                    )
                 )
-            )
 
         return await run_in_threadpool(_graphql_request)
 
+    @contextmanager
+    def wrap_graphql_execution(self, request: Request):
```

Comment:
super nit:   "wrap" feels off to me for this,  maybe `graphql_execution_context` `graphql_request_context` `graphql_request_cm` ? What ever you prefer just throwing some ideas out